src/filters/detectors/assurance.cpp

```diff
@@ -9,71 +10,174 @@
 namespace mesos {
 namespace serenity {
 
-Result<Detection> AssuranceDetector::processSample(
-    double_t in) {
-  this->window.push_back(in);
+void AssuranceDetector::shiftBasePoints() {
```

Comment:
What does the assurance detector do? Do you have reference to some documentation on this?